vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -303,29 +303,32 @@ testCloudInit() {
 
   # Limit this test only to Mariner or Azurelinux 
   if [[ "$os_sku" == "CBLMariner" || "$os_sku" == "AzureLinux" ]]; then
+    echo "Checking if cloud-init.log exists..."
     FILE=/var/log/cloud-init.log
     if test -f "$FILE"; then
-      echo "Check cloud-init log exist."
-      if grep 'WARNING\|ERROR' $FILE; then
-        msg=$(grep 'WARNING\|ERROR' $FILE)
-        err $test "Cloud-init log has WARNING/ERROR: ${msg}."
+      echo "Cloud-init log exist."
+      local grep_output=$(grep 'WARNING\|ERROR' ${FILE})
+      local grep_status=$?
+      if [ ${grep_status} -eq 0 ]; then
+        err $test "Cloud-init log has WARNING/ERROR: ${grep_output}."
       else
         echo "Cloud-init log is OK."
       fi
     else
       err $test "Check cloud-init log does not exist."
     fi
 
-    if cloud-init status --wait; then
+    echo "Checking cloud-init status..."
+    local cloud_init_output=$(cloud-init status --wait)
+    local cloud_init_status=$?
+    if [ ${cloud_init_status} -eq 0 ]; then
```

Comment:
what about error code 2 

https://docs.cloud-init.io/en/latest/explanation/return_codes.html#id1

Are we handling this properly in the official code, (not in the test ?) 
maybe we should distinguish between 0, 1 and 2 ?